add sdpi p history service option (#452)
IHE/DEV.SDPi: asciidoc/volume2/dev-32/tf2-ch-a-mdpws-dev-32.adoc

:var_uri_ws_eventing_notification: https://www.w3.org/Submission/2006/SUBM-WS-Eventing-20060315/#Notifications

:var_label_dev_32_message_subscribe: Subscribe
:var_label_dev_32_message_change_sequence_report: ChangeSequenceReport
:var_uri_dpws_messaging: http://docs.oasis-open.org/ws-dd/dpws/1.1/os/wsdd-dpws-1.1-spec-os.html#_Toc228672084

[#vol2_clause_appendix_mdpws_dev_32]
==== MDPWS: Retrieve Archive Data  [DEV-32]

This section specifies the MDPWS data transmission for messages defined in <<vol2_clause_dev_32>>.

> // ---------- EpisodicAlertReport ---------

Reviewer: remove line?
or clarify

====== Referenced Standards

* <<ref_w3c_ws_eventing_2006>> {var_uri_ws_eventing_subscribe}[Section 3.1 Subscribe]
* <<ref_oasis_dpws_2009>> {var_uri_dpws_eventing}[Section 5 Eventing]

====== Message Outline

.WS-Eventing Subscribe message
[source#vol2_clause_appendix_a_mdpws_dev_32_subscribe,xml]
----
include::../../listings/vol2-clause-appendix-a-mdpws-dev-32-subscribe.xml[]
----

:var_trigger_events_ref: <<vol2_clause_dev_32_message_notification>>
include::../dev-a-default-trigger-events.adoc[]

====== Message Semantics

`s12:Envelope/s12:Body/wse:Subscribe/wse:Filter/@Dialect`:: To be set to `urn:oid:1.3.6.1.4.1.19376.1.6.2.10.1.3.1.1`, which is the default one defined in this specification. Other dialects may be supported but are not in the scope herein.
`s12:Envelope/s12:Body/wse:Subscribe/wse:Filter/hm:ProvideInitialMdib`:: Set to `true` if an initial <<acronym_mdib>> is requested or `false` otherwise.
`s12:Envelope/s12:Body/wse:Subscribe/wse:Filter/hm:LanguageFilter`:: A list of languages for localized texts to be requested (see data type description in <<vol2_clause_appendix_a_mdpws_history_service_xmlschema>> for more details).

> `s12:Envelope/s12:Body/wse:Subscribe/wse:Filter/hm:Range`:: A range that selects the historic <<acronym_mdib>> data, either based on time frames or <<acronym_mdib>> versions (see data type description in <<vol2_clause_appendix_a_mdpws_history_service_xmlschema>> for more details).

Reviewer 2: Do we need to deal with time epochs here? I can see three options:
1. drop support for requesting history by time, or use relative time like subscriptions.
2. leave the provider free to decide which time-reference frame it should use when comparing mdib timestamps and history query time-stamps.
3. add attributes to select epoch. 

1 seems best to me, but I'm not sure what the use cases are for this one. 2 seems okay as well because time is supposed to be UTC time, which is independent of non-slewing time-adjustments. If 2, perhaps a note along the lines : "The provider may not reliably select the correct change reports for time ranges that span an abrupt time adjustment". 

How about: the provider

Reviewer 2: Time is quite useful for testing; saves figuring out all the mdib versions. Otherwise I don't see the use cases. It could be okay to just have a warning that if you use time selection then it might not be perfect. 

Does omitting parts of the range over complicate? I wonder if it could simplify things from the consumer side by not having to figure out what range to provide.

Author: > Time is quite useful for testing; saves figuring out all the mdib versions. 

I agree, but it is also a tough nut if non-slewing adjustments are taken into account - could also be quite problematic for end-users of the system to understand whats happening.

> Does omitting parts of the range over complicate?

The range use-case was mainly intended to be used for backfilling, where a consumer knows exactly which information it missed. I agree that requesting by time is the more prominent use-case if you want to analyze historic data. In the end, answering to time-based requests is always a best guess story. In the end, the user has to inspect the assigned patient and plausibility of the sequence of data that has been returned by the history service.

:var_uri_ws_eventing_notification: https://www.w3.org/Submission/2006/SUBM-WS-Eventing-20060315/#Notifications

:var_label_dev_32_message_subscribe: Subscribe
:var_label_dev_32_message_change_sequence_report: ChangeSequenceReport
:var_uri_dpws_messaging: http://docs.oasis-open.org/ws-dd/dpws/1.1/os/wsdd-dpws-1.1-spec-os.html#_Toc228672084

[#vol2_clause_appendix_mdpws_dev_32]
==== MDPWS: Retrieve Archive Data  [DEV-32]

This section specifies the MDPWS data transmission for messages defined in <<vol2_clause_dev_32>>.

===== {var_label_dev_32_message_subscribe} Message

> <<vol0_transaction_summary_dev_32>> leverages <<vol0_transaction_summary_dev_27>> to initiate a subscription, which is used to deliver historic <<acronym_mdib>> data to a <<vol1_clause_sdpi_p_somds_consumer>>.

Reviewer: ```suggestion
<<vol0_transaction_summary_dev_32>> leverages <<vol0_transaction_summary_dev_27>> to initiate a subscription, which is used to deliver historic <<acronym_mdib>> data to a <<vol1_spec_sdpi_p_actor_somds_consumer>>.
```

The {var_label_dev_32_message_subscribe} message contains filter criteria to the <<term_history_service>>, which is defined in this section. Further message outlines are shown in <<vol2_clause_appendix_mdpws_dev_27>>.

> The {<<vol2_clause_appendix_mdpws_dev_27>>} message is encoded by using {var_uri_dpws_messaging}[DPWS Messaging].

Reviewer: ```suggestion
The <<vol2_clause_appendix_mdpws_dev_27>> message is encoded by using {var_uri_dpws_messaging}[DPWS Messaging].
```

====== Referenced Standards

* <<ref_w3c_ws_eventing_2006>> {var_uri_ws_eventing_subscribe}[Section 3.1 Subscribe]
* <<ref_oasis_dpws_2009>> {var_uri_dpws_eventing}[Section 5 Eventing]

====== Message Outline

.WS-Eventing Subscribe message
[source#vol2_clause_appendix_a_mdpws_dev_32_subscribe,xml]
----
include::../../listings/vol2-clause-appendix-a-mdpws-dev-32-subscribe.xml[]
----

:var_trigger_events_ref: <<vol2_clause_dev_32_message_notification>>
include::../dev-a-default-trigger-events.adoc[]

====== Message Semantics

`s12:Envelope/s12:Body/wse:Subscribe/wse:Filter/@Dialect`:: To be set to `urn:oid:1.3.6.1.4.1.19376.1.6.2.10.1.3.1.1`, which is the default one defined in this specification. Other dialects may be supported but are not in the scope herein.

> `s12:Envelope/s12:Body/wse:Subscribe/wse:Filter/hm:ProvideInitialMdib`:: Set to `true` if an initial <<acronym_mdib>> is requested or `false` otherwise.

Reviewer: here and below it should be 
```suggestion
`s12:Envelope/s12:Body/wse:Subscribe/wse:Filter/hm:HistoryQuery/hm:ProvideInitialMdib`:: Set to `true` if an initial <<acronym_mdib>> is requested or `false` otherwise.
```

:var_expected_actions_ref: <<vol2_clause_dev_32_message_notification>>
include::../dev-a-default-expected-actions.adoc[]

===== Use-cases

This section provides sequence diagrams depicting different <<term_history_service>> use-cases, namely

- xref:vol2_figure_dev_32_usecase_version_range[Requesting a specific range of versions]
- xref:vol2_figure_dev_32_usecase_one_sequence_id[Requesting a time range spanning one <<acronym_mdib>> sequence]
- xref:vol2_figure_dev_32_usecase_two_sequence_ids[Requesting a time range spanning multiple <<acronym_mdib>> sequences]

.<<term_history_service>>: Requesting a specific range of versions
[plantuml#vol2_figure_dev_32_usecase_version_range, target=puml-dev-32-usecase-version-range, format=svg]
....
include::../../plantuml/vol2-figure-dev-32-usecase-version-range.puml[]
....

.<<term_history_service>>: Requesting a time range spanning one <<acronym_mdib>> sequence
[plantuml#vol2_figure_dev_32_usecase_one_sequence_id, target=puml-dev-32-usecase-one-sequence-id, format=svg]
....
include::../../plantuml/vol2-figure-dev-32-usecase-one-sequence-id.puml[]
....

> .<<term_history_service>>: Requesting a time range spanning one <<acronym_mdib>> sequence

Reviewer: ```suggestion
.<<term_history_service>>: Requesting a time range spanning multiple <<acronym_mdib>> sequences
```
